### ISSUE TITLE
3.7/4.x Cake\Form\Form class misses 4.x getters/setters + deprecation

### ISSUE THREAD

**Issue by inoas** [cakephp/cakephp#12596]
This is a (multiple allowed):

* [x] bug
* [x] enhancement

* CakePHP Version: 3.6.x

### What you did
tried to call `\Cake\Form\Form::getErrors()`

**Comment by inoas** [comment]
I am using that - however it is unexpeted/inconsistent with how Entity::getErrors() works.

**Comment by ADmad** [comment]
We could add an alias `Form::getErrors()` for `Form::errors()` in 3.7 and remove `Form::errors()` in 4.0 if there's a consensus for it.

**Comment by markstory** [comment]
Do we need to remove the old method?  For folks who are using the current method it is change for the sake of change.

**Comment by inoas** [comment]
👍 for an alias

**Comment by inoas** [comment]
PR is up.

From the same class:
```
    public function validator(Validator $validator = null)
    {
        deprecationWarning(
            'Form::validator() is deprecated. ' .
            'Use Form::getValidator()/setValidator() instead.'
        );
```

TLDR:
@markstory well I am with you, but what about consistency and purity/non-duplication?

As I said month ago, in general I don't like having things forced into getters+setters but I'd have categorized classes into builders, readers and regular classes, builders dropping their set prefix (but for fetching things having a get prefix), readers dropping their get prefix (but for writing things in case, using a set prefix) and regular classes being the once having set/get prefix everywhere like all our classes (or most) aim to.
